feat: add rails documentation for migration to v8 (#19)
ForestAdmin/documentation: how-tos/maintain/upgrade-notes-rails/upgrade-to-v8.md

  ensure a smooth update.​
---

# Upgrade to v8

This upgrade unlocks the following features:

* Use templating in the filters of Chart components
* Add conditions to your role permissions

## Upgrading to v8

{% hint style="warning" %}
As for any dependency upgrade, it's very important to **test this upgrade** **in your testing environments**. Not doing so could result in your admin panel being unusable.
{% endhint %}

To upgrade to v8, first update your project according to the [_Breaking Changes_](../upgrade-notes-rails/upgrade-to-v8.md#breaking-changes) section below.&#x20;

If you're upgrading from an older version, please make sure you've also read the previous upgrade notes ([v7](upgrade-to-v7.md), [v6](upgrade-to-v6.md),..)

To upgrade to v8, **update the version in your Gemfile**, then run the following and update your project as shown in the _Breaking Changes_ section below.

```javascript
bundle install
```

{% hint style="info" %}
In case of a regression introduced in Production after the upgrade, a rollback to your previous liana version 7 is the fastest way to restore your admin panel.
{% endhint %}

## Breaking changes

> This new version (v8) does not support the old role system. If you are in the case, please follow [this procedure](../migrate-to-the-new-role-system.md) in order to migrate to the new role system **before** you attempt to upgrade to version 8.

Reviewer: ```suggestion
This new version (v8) does not support the old role system. If you are still using the old role system, please follow [this procedure](../migrate-to-the-new-role-system.md) in order to migrate to the new role system **before** you attempt to upgrade to version 8.
```

{% hint style="warning" %}
**How do I know if I'm using the old or new role system?**

If you have access to roles (Project settings > Roles) as designed below...\
\
![](<../../../.gitbook/assets/image (10).png>)\
\

> then you are using the new role system and can go ahead with the upgrade to v9.

Reviewer: ```suggestion
then you are using the new role system and can go ahead with the upgrade to v8.
```

{% endhint %}

> If you were using the ```ForestLiana::PermissionsChecker``` for check the permission on overriding route. 

Reviewer: ```suggestion
In your code, if you were using the ```ForestLiana::PermissionsChecker``` for check the permission on overriding route. 
```
